AMP-111643 Update Taxonomy API docs (#329)
amplitude/amplitude-docs: content/collections/api/en/taxonomy.md

|<div class="big-column">Name</div>|Description|
|-----|---------|
|`event_property`| <span class="required">Required</span>. String. Name of the event property.|

> |`event_type`|<span class="optional">Optional</span>. String. Name of the event type to which the event property belongs to. <ul><li>If the event property already exists on this event type, we will return a `409 Conflict` error.</li><li>If the event property already exists but not on this event type, we will create an override for this property.</li><li>If the event property does not exist anywhere, we will not create an override for this property.</li></ul>|

Author: Weird, I was able to get that working (after struggling for a bit) in my local env.

|`description`|<span class="optional">Optional</span>. String. The event property's description.|
|`type`| <span class="optional">Optional</span>. String. Available with Govern Add-on. The event property's data type. Acceptable values are `string`, `number`, `boolean`, `enum`, and `any`|
|`regex`| <span class="optional">Optional</span>. String. Available with Govern Add-on. Regular expression, custom regex used for pattern matching or more complex values. For example, property zip code must have pattern `[0-9]{5}` <ul><li>Only applicable to the string type.</li></ul>|
|`enum_values`|<span class="optional">Optional</span>. String. Available with Govern Add-on. List of allowed values.|

> |`is_array_type`|<span class="optional">Optional</span>. Boolean. Available with Govern Add-on. <ul><li>Use the `type` parameter to set the type of array elements.</li></ul>|

Reviewer: Please check on whether the Govern add-on is still a thing.

> |`is_array_type`|<span class="optional">Optional</span>. Boolean. Available with Govern Add-on. <ul><li>Use the `type` parameter to set the type of array elements.</li></ul>|

Author: Pretty sure it isn't will remove it.

|<div class="big-column">Name</div>|Description|
|-----|---------|

> |`event_type`|<span class="optional">Optional</span>. Name of the event type to which the event properties belong to. <ul><li>If the event property already exists on this event type, we will return a `409 Conflict` error.</li><li>If the event property already exists but not on this event type, we will create an override for this property.</li><li>If the event property does not exist anywhere, we will not create an override for this property.</li></ul>|

Reviewer: ```suggestion
|`event_type`|<span class="optional">Optional</span>. Name of the event type to which the event properties belong to. If the event property already exists on this event type, Amplitude returns a `409 Conflict` error. If the event property already exists but not on this event type, Amplitude creates an override for this property. If the event property doesn't exist anywhere, Amplitude doesn't create an override for this property.|
```

> |`overrideScope`|<span class="optional">Optional</span>. Determines how we should act on this event property. <ul><li>Only applicable if event_type is provided.</li><li>`overrideScope` is not provided: we will update property override on the event if it exists on the event, or the shared property if no override exists on the event.</li><li>With `overrideScope: "override"`, it will create an override if none exists on the event, then update that overridden property, or it will update the existing override if one already exists.</li><li>With `overrideScope: "shared"` it will remove the property override on the event if one exists on the event, then update the shared property, or it will just update the shared property if no property override exists.</li></ul>|

Reviewer: ```suggestion
|`overrideScope`|<span class="optional">Optional</span>. Determines how we should act on this event property. Only applicable if event_type is present. If `overrideScope` is not present, Amplitude updates property override on the event if it exists on the event, or the shared property if no override exists on the event. With `overrideScope: "override"`, Amplitude creates an override if none exists on the event, then updates that overridden property, or it updates the existing override if one already exists. With `overrideScope: "shared"`, Amplitude removes the property override on the event if one exists on the event, then updates the shared property, or updates the shared property if no property override exists.|
```

|`description`|<span class="optional">Optional</span>. String. The event property's description.|
|`new_event_property_value`|<span class="optional">Optional</span>. String. Available with Govern Add-on. The new name of the event property.|
|`type`| <span class="optional">Optional</span>. String. Available with Govern Add-on. The event property's data type. Acceptable values are `string`, `number`, `boolean`, `enum`, and `any`|
|`regex`|<span class="optional">Optional</span>. String. Available with Govern Add-on. Regular expression, custom regex used for pattern matching or more complex values. For example, property zip code must have pattern `[0-9]{5}` |
|`enum_values`| <span class="optional">Optional</span>. String. Available with Govern Add-on. List of allowed values.|
|`is_array_type`| <span class="optional">Optional</span>. Boolean. Available with Govern Add-on.|
|`is_required`| <span class="optional">Optional</span>. Boolean. Available with Govern Add-on. Marks the property as required.|

> |`classifications`|<span class="optional">Optional</span>. String. List of classifications applicable to this event property. Valid classifications are `PII`, `SENSITIVE` and `REVENUE`. <ul><li>Classifications can only be applied on shared properties and trying to set classifications on an overridden property will result in an error.</li><li>With `overrideScope: "override"`, we return an error because of the same reason mentioned in the previous point.</li></ul> |

Reviewer: ```suggestion
|`classifications`|<span class="optional">Optional</span>. String. List of classifications applicable to this event property. Valid classifications are `PII`, `SENSITIVE` and `REVENUE`. You can only apply classifications on shared properties, and trying to set classifications on an overridden property results in an error. With `overrideScope: "override"`, Amplitude returns an error for the same reason mentioned in the previous point. |
```

}
```

> The Taxonomy API requires an **Enterprise plan**  account or the **Govern add-on** enabled. If your subscription doesn't qualify, the call results in a `404 Bad Request` response.

Reviewer: ```suggestion
The Taxonomy API requires an **Enterprise plan**  account or the **Govern add-on** enabled. If your subscription doesn't qualify, the call results in a `404 Bad Request` response.
```

Reviewer: Are we sure the Govern add-on still exists? I'm pretty sure it doesn't.

{{/partial:tabs}}

{{partial:collapse name="Example: Update a user property"}}

> This example updates a user property called "user_type" to be named "subscription_type", adds a description of "The user's subscription type", and changes the property's data type to `string`. The user property is prefixed with `gp:` in the path because it's a custom user property.

Reviewer: ```suggestion
This example updates the `user_type` user property to be named `subscription_type`, adds a description of "The user's subscription type", and changes the property's data type to `string`. The user property is prefixed with `gp:` in the path because it's a custom user property.
```

|`new_user_property_value`| <span class="optional">Optional</span>. String. New name of the user property type.|
|`description`| <span class="optional">Optional</span>. String. Details to add to the user property type.|
|`type`|<span class="optional">Optional</span>. String. The user property's data type. Acceptable values are `string`, `number`, `boolean`, `enum`, and `any`.|

> |`regex`| <span class="optional">Optional</span>. String. Regular expression or custom regex used for pattern matching and more complex values. For example, 'zip code' property must have pattern `[0-9]{5}`. <ul><li>Only applicable to the `string` type.</li></ul>|

Reviewer: ```suggestion
|`regex`| <span class="optional">Optional</span>. String. Regular expression or custom regex used for pattern matching and more complex values. For example, 'zip code' property must have pattern `[0-9]{5}`. Only applicable to the `string` type.|
```

> |`enum_values`| <span class="optional">Optional</span>. String. List of allowed values, separated by comma. For example: `red, yellow, blue`. <ul><li>Only applicable to the `enum` type.</li></ul>|

Reviewer: ```suggestion
|`enum_values`| <span class="optional">Optional</span>. String. List of allowed values, separated by comma. For example: `red, yellow, blue`. Only applicable to the `enum` type.|
```

> |`is_array_type`|<span class="optional">Optional</span>. Boolean. Specifies whether the property value is an array. <ul><li>Use the `type` parameter to set the type of array elements.</li></ul>|

Reviewer: ```suggestion
|`is_array_type`|<span class="optional">Optional</span>. Boolean. Specifies whether the property value is an array. Use the `type` parameter to set the type of array elements.|
```

        }
    ]
}
```

## Group property

> Group properties are simply properties at the account level. These properties will apply to all users who fall into that account.

Reviewer: ```suggestion
Group properties are properties at the account level. They apply to all users who belong to that account.
```

### Create a group property

Create a group property. Send a `POST` request to the endpoint with the required information in the body.

`POST https://amplitude.com/api/2/taxonomy/group-property`

{{partial:tabs tabs="cURL, HTTP"}}
{{partial:tab name="cURL"}}
```bash
curl --location --request POST 'https://amplitude.com/api/2/taxonomy/group-property' \
-u '{api_key}:{secret_key}' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'group_type=GROUP_TYPE' \
--data-urlencode 'group_property=GROUP_PROPERTY' \
```
{{/partial:tab}}
{{partial:tab name="HTTP"}}
```bash
POST /api/2/taxonomy/group-property HTTP/1.1
Host: amplitude.com
Authorization: Basic {api-key}:{secret-key} #credentials must be base64 encoded
Content-Type: application/x-www-form-urlencoded
Content-Length: 94

group_type=GROUP_TYPE&group_property=GROUP_PROPERTY
```
{{/partial:tab}}
{{/partial:tabs}}

{{partial:collapse name="Example: Create a group property"}}

> This example creates the group property "Group Property 1" with the description "First group property" for the group "Group 1". The group property is a boolean type.

Reviewer: ```suggestion
This example creates the group property `Group Property 1` with the description "First group property" for the group `Group 1`. The group property is a boolean type.
```

{{partial:tabs tabs="cURL, HTTP"}}
{{partial:tab name="cURL"}}
```bash

curl --location --request POST 'https://amplitude.com/api/2/taxonomy/group-property' \
--header 'Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA=' \
--data-urlencode 'group_type=Group 1' \
--data-urlencode 'group_property=Group Property 1' \
--data-urlencode 'description=First Group Property' \
--data-urlencode 'type=boolean'

```
{{/partial:tab}}
{{partial:tab name="HTTP"}}
```bash

POST /api/2/taxonomy/group-property HTTP/1.1
Host: amplitude.com
Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA=

Content-Length: 144

group_type=Group%201&group_property=Group%20Property%201&type=boolean&description=First%20Group%20Property
```
{{/partial:tab}}
{{/partial:tabs}}
{{/partial:collapse}}

#### Body parameters

|<div class="big-column">Name</div>|Description|
|-----|---------|

> |`group_property`| <span class="required">Required</span>. String. Name of the group property. <ul><li>Prefix custom group properties with `grp:`</li></ul>|

Reviewer: ```suggestion
|`group_property`| <span class="required">Required</span>. String. Name of the group property. Prefix custom group properties with `grp:`|
```

> |`group_type`|<span class="optional">Optional</span>. String. Name of the group type the group property belongs to. <ul><li>If the group type does not exist, we will return a `404 Not Found` error.</li><li>If the group property already exists on this group type, we will return a `409 Conflict` error.</li><li>If the group property already exists but not on this group type, we will create an override for this property.</li><li>If the group property does not exist anywhere, we will not create an override for this property.</li><li>If the group property exists and is an Amplitude sourced group property, providing any extra arguments other than `group_property` and `group_type` will result in an error because Amplitude sourced group properties cannot be edited.</li></ul>|

Reviewer: ```suggestion
|`group_type`|<span class="optional">Optional</span>. String. Name of the group type the group property belongs to. If the group type doesn't exist, Amplitude returns a `404 Not Found` error. If the group property already exists on this group type, Amplitude returns a `409 Conflict` error. If the group property already exists but not on this group type, Amplitude creates an override for this property. If the group property doesn't exist anywhere, Amplitude doesn't create an override for this property. If the group property exists and is an Amplitude-sourced group property, providing any extra arguments other than `group_property` and `group_type` results in an error because you can't edit Amplitude-sourced group properties. |
```

|`description`|<span class="optional">Optional</span>. String. The group property's description.|
|`type`| <span class="optional">Optional</span>. String. Data type of the group property. It must be one of `any` (default), `string` (default if array type is true), `number`, `boolean`, `enum`|

> |`regex`| <span class="optional">Optional</span>. String. Regular expression, custom regex used for pattern matching or more complex values. For example, property zip code must have pattern `[0-9]{5}` <ul><li>Only applicable to the string type.</li></ul>|

Reviewer: ```suggestion
|`regex`| <span class="optional">Optional</span>. String. Regular expression, custom regex used for pattern matching or more complex values. For example, property zip code must have pattern `[0-9]{5}` Applies only to the `string` type.|
```

> |`enum_values`|<span class="optional">Optional</span>. String. List of allowed values. <ul><li>Only applicable to the enum type.</li></ul>|

Reviewer: ```suggestion
|`enum_values`|<span class="optional">Optional</span>. String. List of allowed values. Applies only to the `enum` type.|
```

> |`is_array_type`|<span class="optional">Optional</span>. Boolean. Property is an array type. <ul><li>Use the `type` parameter to set the type of array elements.</li></ul>|

Reviewer: ```suggestion
|`is_array_type`|<span class="optional">Optional</span>. Boolean. Property is an array type.  Use the `type` parameter to set the type of array elements. |
```

> |`classifications`|<span class="optional">Optional</span>. String. List of classifications applicable to this group property. Valid classifications are `PII`, `SENSITIVE` and `REVENUE`. <ul><li>Classifications can only be applied on shared properties and trying to set classifications on an overridden property will result in an error.</li></ul> |

Reviewer: ```suggestion
|`classifications`|<span class="optional">Optional</span>. String. List of classifications applicable to this group property. Valid classifications are `PII`, `SENSITIVE` and `REVENUE`. You can only apply classifications on shared properties, and trying to set classifications on an overridden property results in an error. |
```

#### 200 OK response

A successful request returns a `200 OK` status and a JSON body.

```json
{
    "success": true
}
```

#### 409 conflict response

> If there is a problem with your request, the request returns a `409 Conflict` status, and a JSON body with more information.

Reviewer: ```suggestion
If there's a problem with your request, the request returns a `409 Conflict` status and a JSON body with more information.
```

```json
{
    "success": false,
    "errors": [
        {
            "message": "Attempted to add a group property, \"Group Property 1\", that already exists."
        }
    ]
}
```

### Get group properties

> Get shared (or) group specific group properties.

Reviewer: ```suggestion
Get shared or group-specific group properties.
```

--data-urlencode 'group_type=GROUP_TYPE'
```
{{/partial:tab}}
{{partial:tab name="HTTP"}}
```bash

GET /api/2/taxonomy/group-property HTTP/1.1
Host: amplitude.com
Authorization: Basic {api-key}:{secret-key} #credentials must be base64 encoded

group_type=GROUP_TYPE
```
{{/partial:tab}}
{{/partial:tabs}}

{{partial:collapse name="Example: get all of a group's properties"}}
This example gets all group properties for the "Group 1" group type.

{{partial:tabs tabs="cURL, HTTP"}}
{{partial:tab name="cURL"}}
```bash

curl --location --request GET 'https://amplitude.com/api/2/taxonomy/group-property' \
--header 'Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA==' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'group_type=Group 1'

```
{{/partial:tab}}
{{partial:tab name="HTTP"}}
```bash

GET /api/2/taxonomy/group-property HTTP/1.1
Host: amplitude.com
Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA==
Content-Type: application/x-www-form-urlencoded
Content-Length: 26

group_type=Group%201
```
{{/partial:tab}}
{{/partial:tabs}}
{{/partial:collapse}}

#### Body parameters

|<div class="big-column">Name</div>|Description|
|-----|---------|

> |`group_type`|<span class="optional">Optional</span>. Name of the group type.  If `group_type` is present, Amplitude returns all group properties associated with this group type. If `group_type` isn't present, Amplitude returns all shared group properties in your tracking plan.|

Reviewer: ```suggestion
|`group_type`|<span class="optional">Optional</span>. Name of the group type. If `group_type` is present, Amplitude returns all group properties associated with this group type. If `group_type` isn't present, Amplitude returns all shared group properties in your tracking plan.|
```

> |`group_type`|<span class="optional">Optional</span>. Name of the group type. <ul><li>If `group_type` is provided, we return all group properties associated with this group type.</li><li>If `group_type` is not provided, we return all shared group properties in your tracking plan.</li></ul>|

Reviewer: ```suggestion
|`group_type`|<span class="optional">Optional</span>. Name of the group type.  If `group_type` is present, Amplitude returns all group properties associated with this group type. If `group_type` isn't present, Amplitude returns all shared group properties in your tracking plan.|
```

#### 200 OK response

A successful request returns a `200 OK` status and a JSON body with a list of group properties and their data.

```json
{
    "success": true,
    "data": [
        {
            "group_type": "Group 1",
            "group_property": "grp:Group Property 1",
            "description": "First Group Property",
            "type": "string",
            "enum_values": null,
            "regex": null,
            "is_array_type": false,
            "classifications": ["PII"]
        },
        {
            "group_type": "Group 1",
            "group_property": "grp:Group Property 2",
            "description": "Second Group Property",
            "type": "string",
            "enum_values": null,
            "regex": null,
            "is_array_type": false,
            "classifications": []
        },
    ]
}
```

### Get a single group property

> Get a single group property. Send a `GET` request with the group property name as a path parameter, and optionally, the group type in the body.

Reviewer: ```suggestion
Get a single group property. Send a `GET` request with the group property name as a path parameter and, optionally, the group type in the body.
```

`GET https://amplitude.com/api/2/taxonomy/group-property/:group_property`

{{partial:tabs tabs="cURL, HTTP"}}
{{partial:tab name="cURL"}}
```bash
curl --location --request GET 'https://amplitude.com/api/2/taxonomy/group-property/GROUP_PROPERTY' \
-u '{api_key}:{secret_key}' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'group_type=GROUP_TYPE'
```
{{/partial:tab}}
{{partial:tab name="HTTP"}}
```bash
GET /api/2/taxonomy/group-property/GROUP_PROPERTY HTTP/1.1
Host: amplitude.com
Authorization: Basic {api-key}:{secret-key} #credentials must be base64 encoded

group_type=GROUP_TYPE
```
{{/partial:tab}}
{{/partial:tabs}}

{{partial:collapse name="Example: Get a property from a specific group type"}}

> This example gets a group property named "Group Property 1" for the "Group 1" group type.

Reviewer: ```suggestion
This example gets a group property named `Group Property 1` for the `Group 1` group type.
```

{{/partial:tabs}}

{{partial:collapse name="Example: Update a group property"}}
This example updates the "Group Property 1" property. It changes the name to "Group Property 1 - Renamed" and adds a description.

{{partial:tabs tabs="cURL, HTTP"}}
{{partial:tab name="cURL"}}
```bash

curl --location --request PUT 'https://amplitude.com/api/2/taxonomy/group-property/grp:Group Property 1' \
--header 'Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA==' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'description=First Group Property Updated' \
--data-urlencode 'new_group_property_value=grp:Group Property - Renamed' \
```
{{/partial:tab}}
{{partial:tab name="HTTP"}}
```bash

PUT /api/2/taxonomy/group-property/grp:Group Property 1 HTTP/1.1
Host: amplitude.com
Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA==
Content-Type: application/x-www-form-urlencoded
Content-Length: 130

group_type=Group%201&description=First%20Group%20Property%20Updated&new_group_property_value=grp%3AGroup%20Property%201%20Renamed
```
{{/partial:tab}}
{{/partial:tabs}}
{{/partial:collapse}}

#### Path parameters

|<div class="big-column">Name</div>|Description|
|-----|---------|

> |`group_property`|<span class="required">Required</span>. Name of the group property. <ul><li>Prefix custom group properties with `grp:`</li><li>Amplitude sourced group properties (names without the `grp:` prefix) cannot be edited.</li></ul>|

Reviewer: ```suggestion
|`group_property`|<span class="required">Required</span>. Name of the group property.  Prefix custom group properties with `grp:` Amplitude-sourced group properties (names without the `grp:` prefix) aren't editable. |
```

#### Body parameters

|<div class="big-column">Name</div>|Description|
|-----|---------|

> |`group_type`|<span class="optional">Optional</span>. Name of the group type the group property belongs to. <ul><li>If the group type does not exist, we will return a `404 Not Found` error.</li></ul>|

Reviewer: ```suggestion
|`group_type`|<span class="optional">Optional</span>. Name of the group type the group property belongs to.  If the group type doesn't exist, Amplitude returns a `404 Not Found` error. |
```

> |`overrideScope`|<span class="optional">Optional</span>. Determines how we should act on this group property. <ul><li>Only applicable if `group_type` is provided.</li><li>`overrideScope` is not provided: we will update property override on the group type if it exists on the group type, or the shared property if no override exists on the group type.</li><li>With `overrideScope: "override"`, it will create an override if none exists on the group type, then update that overridden property, or it will update the existing override if one already exists.</li><li>With `overrideScope: "shared"` it will remove the property override on the group type if one exists on the group type, then update the shared property, or it will just update the shared property if no property override exists.</li></ul>|

Reviewer: ```suggestion
|`overrideScope`|<span class="optional">Optional</span>. Determines how we should act on this group property.  Only applicable if `group_type` is present. If `overrideScope` is not present, Amplitude updates property override on the group type if it exists on the group type, or the shared property if no override exists on the group type. With `overrideScope: "override"`, Amplitude creates an override if none exists on the group type, then updates that overridden property, or updates the existing override if one already exists. With `overrideScope: "shared"`, Amplitude removes the property override on the group type if one exists on the group type, then updates the shared property, or it updates the shared property if no property override exists.|
```

|`description`|<span class="optional">Optional</span>. String. Description of the group property.|
|`new_group_property_value`|<span class="optional">Optional</span>. String. The new name of the group property.|
|`type`| <span class="optional">Optional</span>. String. Data type of the group property. It must be one of `any` (default), `string` (default if array type is true), `number`, `boolean`, `enum`|

> |`regex`|<span class="optional">Optional</span>. String. Regular expression, custom regex used for pattern matching or more complex values. For example, property zip code must have pattern `[0-9]{5}`  <ul><li>Only applicable to the `string` type.</li></ul>|

Reviewer: ```suggestion
|`regex`|<span class="optional">Optional</span>. String. Regular expression, custom regex used for pattern matching or more complex values. For example, property zip code must have pattern `[0-9]{5}`  Applies only to the `string` type.|
```

|`enum_values`| <span class="optional">Optional</span>. String. List of enum values. <ul><li>Only applicable to the `enum` type.</li></ul>|
|`is_array_type`| <span class="optional">Optional</span>. Boolean. Property is an array type. <ul><li>Use the `type` parameter to set the type of array elements.</li></ul>|

> |`classifications`|<span class="optional">Optional</span>. String. List of classifications applicable to this group property. Valid classifications are `PII`, `SENSITIVE` and `REVENUE`. <ul><li>Classifications can only be applied on shared properties and trying to set classifications on an overridden property will result in an error.</li><li>With `overrideScope: "override"`, we return an error because of the same reason mentioned in the previous point.</li></ul> |

Reviewer: ```suggestion
|`classifications`|<span class="optional">Optional</span>. String. List of classifications applicable to this group property. Valid classifications are `PII`, `SENSITIVE` and `REVENUE`. <ul><li>You can only apply classifications on shared properties, and trying to set classifications on an overridden property results in an error. With `overrideScope: "override"`, Amplitude returns an error for the same reason mentioned in the previous point. |
```
